### ISSUE TITLE
[bug] Signature decoding error: Encoded text cannot have a 6-bit remainder.

### ISSUE THREAD

**Issue by mrquantumoff** [tauri-apps/tauri#5142]
### Describe the bug

My updater doesn't work. I generated keys as it was shown in [the updater guide](https://tauri.app/v1/guides/distribution/updater/#signing-updates), but instead on release builds it just gets stuck on installUpdate() phase and when I try to reproduce the issue in debug builds by reducing the version in tauri.conf.json I get this error in js console: 
![image](https://user-images.githubusercontent.com/36820091/188564729-79844fa0-4cd0-4363-b14e-304288d8b19c.png)

### Reproduction

1. Create new keypair
2. Use them in the updater

### Expected behavior

Updater works

### Platform and versions

Environment
  › OS: Windows 10.0.22622 X64
  › Webview2: 105.0.1343.27
  › MSVC: 
      - Visual Studio Community 2022
      - Visual Studio Build Tools 2019
  › Node.js: 16.17.0
  › npm: 8.16.0
  › pnpm: 7.11.0
  › yarn: Not installed!
  › rustup: 1.25.1
  › rustc: 1.63.0
  › cargo: 1.63.0
  › Rust toolchain: stable-x86_64-pc-windows-msvc 

Packages
  › @tauri-apps/cli [NPM]: 1.0.5
  › @tauri-apps/api [NPM]: 1.0.2
  › tauri [RUST]: 1.0.5,
  › tauri-build [RUST]: 1.0.4,
  › tao [RUST]: 0.12.2,
  › wry [RUST]: 0.19.0,

App
  › build-type: bundle
  › CSP: unset
  › distDir: ../dist
  › devPath: http://localhost:1420/
  › framework: React

App directory structure
  ├─ .git
  ├─ .github
  ├─ .tauri
  ├─ .vscode
  ├─ dist
  ├─ node_modules
  ├─ public
  ├─ src
  └─ src-tauri

**Comment by lucasfernog** [comment]
Either your public key or the signature is wrong. Can you share what values you picked from a fresh `tauri signer generate` usage? Also make sure the signature contents is exactly what's in the `.sig` files.

**Comment by mrquantumoff** [comment]
![image](https://user-images.githubusercontent.com/36820091/188629815-09c716b8-2b6c-4b94-a571-9d75101923e3.png)

**Comment by mrquantumoff** [comment]
I have verified the signatures and I found another problem (signatures weren't the same).
But, now I have another error saying "Signature decoding error: Invalid byte 58, offset 5."

**Comment by mrquantumoff** [comment]
I'm gonna open another issue addressing this bug

**Comment by Levminer** [comment]
Hi, @mrquantumoff did you open a new issue? I'm getting the same error

![image](https://user-images.githubusercontent.com/33373714/189519604-a4a6a502-0dcc-491f-b7ba-d49f888da5fc.png)

**Comment by mrquantumoff** [comment]
Check your update server/file, the signature field is the content of generated .sig file (it was not specific enough in the docs so I opened a pr. It already got merged)

**Comment by Levminer** [comment]
> Check your update server/file, the signature field is the content of generated .sig file (it was not specific enough in the docs so I opened a pr. It already got merged)

Thank you!

**Comment by mrquantumoff** [comment]
You're welcome

**Comment by ankitvermaAT** [comment]
Hi @mrquantumoff, can  you please help me out because I am having the same issue
while I had put environment variable value TAURI_PRIVATE_KEY and public key in tauri.config.json
still getting the same error, Please help to fix this?

**Comment by mrquantumoff** [comment]
check my previous replies, there's the answer on how I solved the issue

**Comment by ankitvermaAT** [comment]
your means: 
 Check your update server/file, the signature field is the content of generated .sig file
 (it was not specific enough in the docs so I opened a pr. It already got merged)

But I can't able to understand, could you please detail me out some more explanation.

**Comment by mrquantumoff** [comment]
your server file must contain the contents of the generated signature file, not a link to them or anything else

**Comment by ankitvermaAT** [comment]
but where is the server file located, kindly help because I am very new in Tauri

**Comment by mrquantumoff** [comment]
you have to create it yourself

**Comment by ankitvermaAT** [comment]
could you please suggest how to do it and what content will be in that file

**Comment by mrquantumoff** [comment]
https://tauri.app/v1/guides/distribution/updater

**Comment by ankitvermaAT** [comment]
While I am using below format in server file:
<?php 
$json = [
  "version"=> "1.0.2",
  "url"=> "https://domain/windows/x86_64/1.0.1/myapp_1.0.2_x64_en-US.msi.zip",
  "signature"=> "https://domain/windows/x86_64/1.0.1/myapp_1.0.2_x64_en-US.msi.zip.sig"
];
echo json_encode($json);
die;
?>
and it returns json object with keys (version, ulr, signature) is this correct?

**Comment by ankitvermaAT** [comment]
Hi @mrquantumoff,  firstly a big thanks for your reply.

I found a mistake i.e. I was using .sig file file in server file instead of it's value.

Now I am getting another error during updater (**Error: Verify signature error: UnexpectedKeyId**) if you have any idea about this error kindly suggest I am following below steps..

1. run : npm run tauri signer generate

2. Got public and private keys

3. Put public key in tauri.config.json (pubkey) node

4. Put private key in env file

5. run : npm run tauri build

6. Got update build (myapp_1.0.2_x64_en-US.msi.zip) and .mis.zip.sig files
Upload .mis.zip on server and update signature value in server json which is inside (.mis.zip.sig)

Now I have my old version build i.e. 1.0.1 run this app and got that error (**Verify signature error: UnexpectedKeyId**)

I had used below packages:

"dependencies": {
"@tauri-apps/api": "^1.5.2",
"axios": "^1.6.7",
"react": "^18.2.0",
"react-dom": "^18.2.0",
"react-router-dom": "^6.22.1"
},
"devDependencies": {
"@tauri-apps/cli": "^1.5.8",
"@types/react": "^18.2.15",
"@types/react-dom": "^18.2.7",
"@vitejs/plugin-react": "^4.2.1",
"typescript": "^5.0.2",
"vite": "^5.0.0"
}
